app/views/atLeastOneDirectorHasNino.scala.html

```diff
@@ -43,3 +43,7 @@
         @submitButton()
     }
 }
+
+@{
+    // $COVERAGE-OFF$
```

Comment:
Can we not just turn off scoverage for all the views if we are gonna do this :p

Reply:
If you do that, then you wouldn't know if things aren't covered in the actual View. This only turns it off for everything below this point in the file (which is the auto-generated code when the Scala compilation occurs).

So, it's better to keep it on and then we know which views haven't been tested :)

Reply:
That seems exceptionally janky. Feels like we are forcing through fake test coverage either way :/

Reply:
Nah, it's real coverage. For the actually view code that's written. The janky bit is just the way the Scala view templates are compiled and generated unaccessible code.

Reply:
If you suppress the `view` package entirely then you'll have no check for coverage at all.

Reply:
Yeah, but throwing those everywhere just to bring it up to a level feels a bit messy for no actual gain on our part

Reply:
![Screenshot 2022-10-07 at 15 32 04](https://user-images.githubusercontent.com/16954819/194579148-7e825542-6adc-4c03-8e0a-d23e138dccf5.png)

Reply:
☝️ the coverage tooling would add instruments for this but its code that can't be accessed - so, the coverage off suppresses that from instrumentation but allows the view to still be checked.

I know it's not really perfect, but, at least if you see a view at 0%, you know there's a. ViewSpec and and ControllerSpec either missing or with missing tests. 🤷‍♂️

Reply:
Yah! I get that but what does adding these lines of code give us but a vanity metric, I don't think it really offers anything to the app or to the testing. If we are teetering on the edge wouldn't increasing the coverage on untested actual code be better than adding extra code to hide the generated code that happens in every service?

Reply:
Well, it's more that it artificially brings your coverage down. Before, this repository excluded all view files from being included in coverage reports - that's more of an issue than this change.

Adding this in actually brought the coverage down initially, and I then added extra tests for things which weren't covered.

So I don't think it's vanity testing if it finds that 3 of the views aren't tested at all when you run the coverage report? It highlights an issue with the testing at either UT/IT level for those views (or, highlights view which exists but aren't actually used anywhere anymore and can be tidied away). 

All these 3 lines do is prevent the coverage tooling reporting on something which is completely out of our control.

Suppressing for the entire package like it did before seems lazy and much worse.

Reply:
I'm not advocating for adding back the entire view package I originally thought that's what those lines were doing so suggested making it a bit cleaner. I'm saying that hiding the generated code specifically isn't really offering us anything but a bit of extra mess in the views I'd rather take the lower percent and bring it up elsewhere unless this is something that is standard across the centre